File: LinkedList/java_linked_list/LinkedList/LinkedList.java
package java_linked_list.LinkedList; 

/**
 * @classname : LinkedList
 */

public class LinkedList {
    
    private LinkedListNode head; 

    public LinkedList() {
        this.head = null; 
    }

    public void add(int data) {
        if(this.head == null) {
            this.head = new LinkedListNode(data); 
        } else {
            LinkedListNode new_node = new LinkedListNode(data), current = this.head;
            while(current.getNext() != null) {
                current = current.getNext(); // iterate over next node. 
            }
            current.setNext(new_node); // add node in List. 
        }
    }

    public void remove(int data) {
        if(this.head != null) {
            LinkedListNode current = this.head, previous = this.head; 
            while(current != null && (current.getData() != data)) {
                previous = current; 
                current = current.getNext(); 
            }
            if(current != null) { // node found in Linked List. 
                if(current == this.head) { // if head is our targe node delete it and update the head value.
                    this.head = this.head.getNext(); // delete head. 
                } else {
                    previous.setNext(current.getNext()); // delete target node. 
                }
            }
        }
    }

    public boolean find(int data){
        if(this.head == null) { // if linkedlist is empty return null. 
            return false; 
        } else {
            LinkedListNode current = this.head; 
            while(current != null) { // iterate untill found the end of linked list. 
                if(current.getData() == data) { // if data found 
                    return true; // return true. 
                }  
                current = current.getNext();  // iterate over next node. 
            } 
            return false; // return false if not found.  
        }
    } 

    public void printList() {
        if(this.head != null) {
            LinkedListNode current = this.head; 
            while(current != null) {
                System.out.print(current.getData() + " "); 
                current = current.getNext(); // iterate over next node. 
            }
            System.out.println(); 
        }       
    }

    public LinkedListNode getHead() {
        return this.head; 
    }

    public void setHead(LinkedListNode new_head) { // funtion to updated head of linked list. 
        this.head = new_head; // update head. 
    }

} // @class : LinkedList. 

File: LinkedList/python_linked_list/LinkedList/LinkedList.py
# filename : LinkedList.py 

# class : LinkedList 

from .LinkedListNode import LinkedListNode  

class LinkedList: 

    def __init__(self): 
        self.head = None # head of linked list. 

    def add(self, element): 
        if self.head is None: 
            self.head = LinkedListNode()
            self.head.data = element 
        else: 
            current = self.head 
            while current.next_link is not None: 
                current = current.next_link 
            current.next_link = LinkedListNode() # add new node in linked list. 
            current.next_link.data = element # update data in added node. 
    
    def printList(self): 
        if self.head is None: 
            print("Linked List is empty")
        else: 
            current = self.head 
            while current is not None: 
                print(f"{current.data}", end = ", ")
                current = current.next_link # iterate to next node. 
            print()
    
    def remove(self, element):
        if self.head is not None: 
            current = self.head 
            previous = current 
            while (current is not None) and (current.data !=  element): 
                previous = current 
                current = current.next_link # iterate to next node. 
            if current is not None: 
                if current is self.head: 
                    self.head = self.head.next_link # update head of linked list, delete the head node. 
                else: 
                    previous.next_link = current.next_link # delete target node. 

    def search(self, key): 
        if self.head is not None: 
            current = self.head 
            while current is not None: 
                if current.data == key: # if key found 
                    return (True, current.data) # return True and data 
                current = current.next_link # iterate over next node. 
            return (False, None) # return False , if key not found. 
    


    